app/components/route-details-header/route-details-header.tsx

```diff
@@ -89,27 +91,32 @@ export interface RouteDetailsHeaderProps {
    */
   screenName?: "routeDetails" | "activeRide"
   style?: ViewStyle
+  eventConfig?: AddCalendarEvent.CreateOptions
 }
 
 export const RouteDetailsHeader = observer(function RouteDetailsHeader(props: RouteDetailsHeaderProps) {
-  const { originId, destinationId, screenName, style } = props
+  const { originId, destinationId, screenName, style, eventConfig } = props
   const { favoriteRoutes } = useStores()
   const navigation = useNavigation()
 
+  const addToCalendar = (eventConfig) => {
+    AddCalendarEvent.presentEventCreatingDialog(eventConfig);
+  }
```

Comment:
On iOS you should ask for permission before adding events to the calendar

Reply:
There's no alert if the user disabled calendar access and clicked on the button. We should add one